### PR TITLE
GO-2238 Fix non-unicode file names conversion

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core/block/import/source/zip.go [hunk 1]
```diff
@@ -27,16 +30,36 @@ func (z *Zip) Initialize(importPath string) error {
...
+		bestGuess, err := chardet.NewTextDetector().DetectBest([]byte(f.Name))
+		if err == nil {
+			open, err := iconv.Open("UTF-8", bestGuess.Charset)
+			defer open.Close()
+			if err == nil {
+				fileName = open.ConvString(f.Name)
+			}
+		}
+		if fileName == "" {
+			fileName = fmt.Sprintf("import file %d", index+1) + filepath.Ext(f.Name)
```
Comment (deff7) [inline core/block/import/source/zip.go:55]: maybe put all string formatting into Sprintf?

File: core/block/import/source/zip.go [hunk 2]
```diff
@@ -27,16 +30,36 @@ func (z *Zip) Initialize(importPath string) error {
...
 	z.fileReaders = fileReaders
 	return nil
 }
 
+func normalizeName(f *zip.File, index int) string {
+	fileName := ""
+	if f.NonUTF8 {
+		bestGuess, err := chardet.NewTextDetector().DetectBest([]byte(f.Name))
+		if err == nil {
+			open, err := iconv.Open("UTF-8", bestGuess.Charset)
```
Comment (deff7) [inline core/block/import/source/zip.go:48]: we need to test it on windows

File: core/block/import/source/zip.go [hunk 3]
```diff
@@ -27,16 +30,36 @@ func (z *Zip) Initialize(importPath string) error {
...
 		}
-		fileReaders[f.Name] = f
+		fileReaders[normalizeName(f, i)] = f
 	}
 	z.fileReaders = fileReaders
 	return nil
 }
 
+func normalizeName(f *zip.File, index int) string {
+	fileName := ""
```
Comment (deff7) [inline core/block/import/source/zip.go:44]: prefer `var fileName string` declaration for empty values

File: core/block/import/source/zip.go (hunk 2 above)
Comment (fat-fellow) [reply core/block/import/source/zip.go:48]: Ok, I will arrange it

**Review by deff7 — APPROVED** [review]

**Pull request merged**